src/app/InteractionModelEngine.cpp

```diff
@@ -490,12 +545,13 @@ Protocols::InteractionModel::Status InteractionModelEngine::OnReadInitialRequest
             }
 
             EventPathIBs::Parser eventpathListParser;
-            err = subscribeRequestParser.GetEventRequests(&eventpathListParser);
+            err                    = subscribeRequestParser.GetEventRequests(&eventpathListParser);
+            bool hasValidEventPath = false;
             if (err == CHIP_NO_ERROR)
             {
-                TLV::TLVReader pathReader;
-                eventpathListParser.GetReader(&pathReader);
-                ReturnErrorCodeIf(TLV::Utilities::Count(pathReader, requestedEventPathCount, false) != CHIP_NO_ERROR,
+                auto subjectDescriptor = apExchangeContext->GetSessionHandle()->AsSecureSession()->GetSubjectDescriptor();
+                ReturnErrorCodeIf(ParseEventPaths(subjectDescriptor, eventpathListParser, hasValidEventPath,
```

Comment:
Can you go up to the corresponding code for attribute paths above (line 535) and make it use the ReturnErrorCodeIf macro as well, so they're both the same?